src/fight/shooting.rs

```diff
@@ -0,0 +1,284 @@
+//! # Shooting phase Module
+//!
+//! This module contain all the functions to emulate the shooting between 2 Regiment
+//! This may evolve to multiple Regiment against multiple Regiment
+
+use crate::{
+    fight::shooting_tools, fight::AttackPosition, fight::ComputeCase, fight::FightCase,
+    fight::FightPredictionResult, regiment,
+};
+
+/// Make two units shot and return the probability that the fight occurs during the game
+///
+/// # Parameters
+/// attackers (&mut regiment::Regiment, &ComputeCase): The attacking regiment
+///
+/// defender (&mut regiment::Regiment, &ComputeCase): The defending regiment
+///
+/// # Return
+/// f64: The probability that the shoot computed occurs
+fn apply_shoot(
+    attacker: (&mut regiment::Regiment, &ComputeCase),
+    defender: (&mut regiment::Regiment, &ComputeCase),
+) -> f64 {
+    let first_damages: (usize, f64) =
+        shooting_tools::compute_case_shooting(attacker.0, defender.0, attacker.1);
+    let second_damages: (usize, f64) =
+        shooting_tools::compute_case_shooting(defender.0, attacker.0, defender.1);
+
+    if attacker.0.get_model().is_banner_bearer() {
+        attacker.0.earn_points(1);
+    }
+    if defender.0.get_model().is_banner_bearer() {
+        defender.0.earn_points(1);
+    }
+    defender.0.take_damage(first_damages.0);
+    attacker.0.earn_points(first_damages.0);
+
+    if defender.0.get_regiment_health_points() > 0 {
+        attacker.0.take_damage(second_damages.0);
+        defender.0.earn_points(second_damages.0);
+        first_damages.1 * second_damages.1
+    } else {
+        first_damages.1
+    }
+}
+
+/// Create a prediction for a fight between two units
+///
+/// # Parameters
+/// attacking_position (AttackPosition): The position of the attacking Regiment
+///
+/// attacking_regiment (&regiment::Regiment): The first unit
+///
+/// defending_regiment (&regiment::Regiment): The second unit
+///
+/// case (&ComputeCase): The scenario from first unit point of view
+///
+/// # Return
+/// FightCase: The case of the fight
+fn create_prediction_shoot(
```

Comment:
During a shooting phase there is no score unless the defending unit lose at least 25% and in this case there is a panic test.